lib/service_actor/arguments_validator.rb

```diff
@@ -3,6 +3,14 @@
 module ServiceActor::ArgumentsValidator
   module_function
 
+  def validate_origin_name(name, origin:)
+    return unless ServiceActor::Result.instance_methods.include?(name.to_sym)
+
+    raise ArgumentError, <<~TXT
```

Comment:
Are not these actors already broken? If someone uses `object_id` he will get unexpected outputs after updating dependencies (result's object_id instead of what has actually been set via `self.object_id`). Warnings might be noticed post facto

Also, I'm sorry that I didn't see these ostruct removal side-effects coming :( Seems like thing are a little bit complicated

Reply:
Maybe we should temporary rollback `Result`'s commits and release warnings first?

Reply:
Hm, actually after [re-inheriting ](https://github.com/sunny/actor/pull/138/commits/31bdaff3f30a94fba717dd4c87a68209c437a6f6)from `BasicObject` the situtation seems to be much better:

```
[1] pry(main)> ServiceActor::Result.instance_methods
=> [:merge!,
 :send,
 :key?,
 :delete,
 :pretty_print,
 :failure?,
 :inspect,
 :fail!,
 :object_id,
 :class,
 :success?,
 :[]=,
 :[],
 :kind_of?,
 :is_a?,
 :respond_to?,
 :to_h,
 :equal?,
 :!,
 :__send__,
 :==,
 :!=,
 :__binding__,
 :instance_eval,
 :instance_exec,
 :__id__]
```

Method `delete` seems to be dangerous, but we can simoky rename/remove it